geometry_msgs/Pose[] poses
string[] colors